### ISSUE TITLE
feat: Complete Evo 2 model integration for DNA sequence classification

### ISSUE THREAD

**Issue by mainuddinMains** [oss-slu/baio#179]
## Summary

Evo 2 model selection exists in the UI and backend but is **non-functional**. When a user selects "Binary (Evo 2 - GPU Required)", the system silently falls back to RandomForest without any warning. This issue tracks all the work needed to make Evo 2 fully operational.

---

### What exists
- `binary_classifiers/evo2_embedder.py` — `Evo2Embedder` class is implemented (uses HuggingFace transformers + mean-pooling of hidden states)
- `metaseq/evo2_client.py` — NVIDIA API client for Evo 2-40B
- `backend/app/services/classification.py` — model resolution logic routes "evo" → `Evo2`
- `frontend/src/components/ConfigPanel.tsx` — UI dropdown includes "Binary (Evo 2 - GPU Required)"
- Training notebooks in `binary_classifiers/training_scripts/`:
  - `evo2_randon_forest_classifier.ipynb`
  - `evo2_support_vector_machine_classifier.ipynb`

### What is missing
- `binary_classifiers/models/evo2_classifier.pkl` — **trained model artifact does not exist**
- No published accuracy benchmarks for Evo 2
- No user-facing warning when Evo 2 falls back to RandomForest
- Evo 2 imports crash test collection on CPU-only environments (documented in `docs/issues/stabilize-evo2-test-ci.md`)

### Confirmed fallback behavior
The following test proves Evo 2 always falls back today:
```python
# tests/test_predict_class_extra.py
def test_evo2_initialization_falls_back_to_random_forest(self):
    predictor = PredictClass(model_name="Evo2")
    assert predictor.model_name == "RandomForest"  # always falls back
```

---

## Current Model Accuracy (RandomForest + 6-mers)

| Metric | Score |
|--------|-------|
| Cross-validation accuracy | 0.9575 |
| Test accuracy | **0.71** |
| ROC AUC | 0.9012 |
| Virus recall | ~0.44 |

There is significant overfitting and poor viral recall. Evo 2 is expected to improve this substantially.

### Expected improvement with transformer-based models

| Model | Expected Test Accuracy | Expected Virus Recall |
|-------|----------------------|-----------------------|
| RandomForest + 6-mers (current) | 0.71 | ~0.44 |
| DNABERT-2 fine-tuned | ~0.88–0.95 | ~0.85–0.92 |
| Nucleotide Transformer | ~0.90–0.96 | ~0.90–0.95 |
| Evo 2 (target) | TBD | TBD |

---

### 1. Train and export the Evo 2 classifier
- [ ] Run `binary_classifiers/training_scripts/evo2_randon_forest_classifier.ipynb` on a GPU machine
- [ ] Run `binary_classifiers/training_scripts/evo2_support_vector_machine_classifier.ipynb`
- [ ] Evaluate both and pick the better-performing one
- [ ] Save the final artifact as `binary_classifiers/models/evo2_classifier.pkl`
- [ ] Document the training dataset, hyperparameters, and hardware used

### 2. Benchmark and document accuracy
- [ ] Run `binary_classifiers/evaluation.py` against the trained Evo 2 model
- [ ] Record: Accuracy, Precision, Recall, F1, ROC AUC, confusion matrix
- [ ] Compare against RandomForest and SVM baselines
- [ ] Add results to `docs/transformer_classification_guide.md`

### 3. Fix CPU-only / CI environment crashes
- [ ] Move `torch` and `transformers` to optional extras in `pyproject.toml` (e.g. `pip install baio[evo2]`)
- [ ] Guard all Evo 2 imports with `try/except ImportError` so test collection does not fail on CPU runners
- [ ] Update GitHub Actions workflow to skip Evo 2 tests unless the `[evo2]` extra is installed

### 4. Add user-facing fallback warning
- [ ] When Evo 2 is selected but falls back to RandomForest, return a warning field in the API response
- [ ] Display the warning in the frontend so the user knows which model was actually used

### 5. Update documentation
- [ ] Add Evo 2 GPU requirements to `README.md` (minimum VRAM, CUDA version)
- [ ] Document how to run the training notebooks
- [ ] Update the Docker guide to mention GPU passthrough (`deploy.resources.reservations.devices`) if Evo 2 is needed inside Docker

---

## Acceptance Criteria

- [ ] Selecting "Binary (Evo 2 - GPU Required)" on a GPU machine uses Evo 2 — not RandomForest
- [ ] Accuracy metrics for Evo 2 are documented and exceed the current 0.71 test accuracy baseline
- [ ] On a CPU-only machine, the fallback to RandomForest shows a clear warning to the user
- [ ] CI passes on CPU runners without Evo 2 dependency errors
- [ ] README documents the GPU requirement for Evo 2

---

## Notes

- Evo 2 requires a CUDA-capable GPU. It will not run on CPU or Apple Silicon MPS without significant code changes.
- The NVIDIA API path (`metaseq/evo2_client.py`) is an alternative for teams without local GPU access — this could be explored as a lower-barrier option.
